modules/api/src/main/java/org/apache/ignite/app/IgnitionProcessor.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.app;
+
+import java.util.ServiceLoader;
+
+/**
+ * Service loader based implementation of an entry point for handling grid lifecycle.
+ */
+public enum IgnitionProcessor implements Ignition {
```

Comment:
What is the point in having service loader based implementation ?

Reply:
It seems to be the best option to load IgnitionImpl that is part of ignite-runner from within ignite-api module. Cause we don't want to have circular references it's not possible for ignite-api to depend on ignite-runner.